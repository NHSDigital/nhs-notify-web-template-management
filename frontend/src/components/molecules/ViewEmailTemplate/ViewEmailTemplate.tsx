'use client';

import { PreviewTemplate } from '@molecules/PreviewTemplate';
import {
  EmailTemplate,
  PageComponentProps,
} from 'nhs-notify-web-template-management-utils';
import { getBasePath } from '@utils/get-base-path';
import { renderEmailMarkdown } from '@utils/markdownit';
import { BackLink } from 'nhsuk-react-components';
import content from '@content/content';
import Link from 'next/link';
import { NHSNotifyMain } from '@atoms/NHSNotifyMain/NHSNotifyMain';

export function ViewEmailTemplate({
  initialState,
}: Readonly<PageComponentProps<EmailTemplate>>) {
  const templateSubjectLine = initialState.subject;
  const templateMessage = initialState.message;

  const html = renderEmailMarkdown(templateMessage);

  const { cannotEdit, createNewTemplate } =
    content.components.viewSubmittedTemplate;

  return (
    <>
      <BackLink href={`${getBasePath()}/manage-templates`}>
        Back to all templates
      </BackLink>
      <NHSNotifyMain>
        <div className='nhsuk-grid-row'>
          <div className='nhsuk-grid-column-full'>
            <PreviewTemplate.Email
              template={initialState}
              subject={templateSubjectLine}
              message={html}
            />
            <p>{cannotEdit}</p>
            <p>{createNewTemplate}</p>
            <p>
              <Link href='/manage-templates'>Back to all templates</Link>
            </p>
          </div>
        </div>
      </NHSNotifyMain>
    </>
  );
}
