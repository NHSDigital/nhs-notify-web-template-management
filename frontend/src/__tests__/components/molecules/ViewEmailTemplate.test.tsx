import { render } from '@testing-library/react';
import { ViewEmailTemplate } from '@molecules/ViewEmailTemplate/ViewEmailTemplate';
import { EmailTemplate } from 'nhs-notify-web-template-management-utils';

describe('ViewEmailTemplate component', () => {
  it('matches submitted snapshot', () => {
    const container = render(
      <ViewEmailTemplate
        initialState={
          {
            id: 'template-id',
            templateType: 'EMAIL',
            name: 'Example template',
            templateStatus: 'SUBMITTED',
            subject: 'Example subject',
            message: 'Example message',
            createdAt: '2025-03-28T12:30:54.684Z',
            updatedAt: '2025-03-28T12:31:54.684Z',
            lockNumber: 1,
          } satisfies EmailTemplate
        }
      />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
