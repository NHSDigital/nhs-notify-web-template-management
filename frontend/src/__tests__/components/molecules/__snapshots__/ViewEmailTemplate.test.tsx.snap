// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewEmailTemplate component matches submitted snapshot 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-back-link"
  >
    <a
      class="nhsuk-back-link__link"
      href="/templates/manage-templates"
    >
      <svg
        aria-hidden="true"
        class="nhsuk-icon nhsuk-icon__chevron-left"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z"
        />
      </svg>
      Back to all templates
    </a>
  </div>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <div
      class="nhsuk-grid-row"
    >
      <div
        class="nhsuk-grid-column-full"
      >
        <h1
          class="preview__heading"
          data-testid="preview-message__heading"
        >
          Example template
        </h1>
        <div
          class="nhsuk-width-container nhsuk-u-margin-bottom-6 nhsuk-body-m"
        >
          <div
            class="preview"
          >
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Template ID
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                template-id
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Type
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              />
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Status
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <strong
                  class="nhsuk-tag nhsuk-tag--grey"
                >
                  Submitted
                </strong>
              </div>
            </div>
          </div>
          <div
            class="nhsuk-u-margin-top-4 preview"
          >
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                  data-testid="preview__heading-0"
                  id="preview-heading-subject"
                >
                  Subject
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <div
                  class="preview__col_content"
                  data-testid="preview__content-0"
                  id="preview-content-subject"
                >
                  Example subject
                </div>
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                  data-testid="preview__heading-1"
                  id="preview-heading-message"
                >
                  Message
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <div
                  class="preview__col_content"
                  data-testid="preview__content-1"
                  id="preview-content-message"
                >
                  <p>
                    Example message
                  </p>
                  

                </div>
              </div>
            </div>
          </div>
        </div>
        <p>
          This template cannot be edited because it has been submitted.
        </p>
        <p>
          If you want to change a submitted or live template, you must create a new template to replace it.
        </p>
        <p>
          <a
            href="/manage-templates"
          >
            Back to all templates
          </a>
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
